packages/tools/src/utilities/math/polyline/polylineHelper.ts

```diff
@@ -0,0 +1,201 @@
+import type { Types } from '@cornerstonejs/core';
+// Epsilon for floating point comparisons
+export const EPSILON = 1e-7; // A bit larger than glMatrix.EPSILON for robustness
+
+// --- Basic Vector and Point Math ---
+export function vec2Subtract(
+  out: Types.Point2,
+  a: Types.Point2,
+  b: Types.Point2
+): Types.Point2 {
+  out[0] = a[0] - b[0];
+  out[1] = a[1] - b[1];
+  return out;
+}
+
+export function vec2CrossZ(a: Types.Point2, b: Types.Point2): number {
+  return a[0] * b[1] - a[1] * b[0];
+}
+
+export function distanceSquared(p1: Types.Point2, p2: Types.Point2): number {
+  const dx = p1[0] - p2[0];
+  const dy = p1[1] - p2[1];
+  return dx * dx + dy * dy;
+}
+
+export function pointsAreEqual(p1: Types.Point2, p2: Types.Point2): boolean {
+  return Math.abs(p1[0] - p2[0]) < EPSILON && Math.abs(p1[1] - p2[1]) < EPSILON;
+}
+
```

Comment:
please use gl-matrix for these 

vec2.subtract
vec2.cross
vec2.squareDistance
vec2.equals